crypto/src/key/mod.rs

```diff
@@ -133,19 +153,34 @@ impl PublicKey {
     }
 
     pub fn verify_message(&self, signature: &Signature, msg: &[u8]) -> bool {
-        let PublicKeyHolder::RistrettoSchnorr(k) = &self.pub_key;
-        match signature {
-            RistrettoSchnorr(s) => k.verify_message(s, msg),
+        match &self.pub_key {
+            PublicKeyHolder::Secp256k1Schnorr(ref k) => match signature {
+                Secp256k1Schnorr(s) => k.verify_message(s, msg),
+                _ => panic!("Wrong key/signature combination"),
```

Comment:
I'm just curious if it is OK to panic here.

Reply:
No. I think it's not the right thing to do.

However, whether we return a `Result<bool>` or just `false` is an open question. I'm tempted to use `Result<bool>` and then allow the dev to use `unwrap_or(false)`, but the issue here is that we're gonna later remove the RistrettoSchnorr in a subsequent PR, so this won't be useful and the panic will disappear anyway because we'll go back to having one key type.